packages/react/src/components/stepper/Stepper.stories.tsx

```diff
@@ -0,0 +1,663 @@
+            invalid={state.fields.firstName.value.length === 0 && state.fields.firstName.visited === true}
+            errorText={
+              state.fields.firstName.value.length === 0 &&
+              state.fields.firstName.visited === true &&
+              'First name is compulsory'
+            }
+            value={state.fields.firstName.value}
+            onChange={(event) =>
+              dispatch({ type: 'changeField', fieldName: 'firstName', newValue: event.target.value })
+            }
+          />
+        )}
+        {state.activeStep === 2 && (
+          <TextInput
+            style={{ width: '300px', paddingTop: '60px' }}
+            id="lastName"
+            label="Last name *"
+            invalid={state.fields.lastName.value.length === 0 && state.fields.lastName.visited === true}
+            errorText={
+              state.fields.lastName.value.length === 0 &&
+              state.fields.lastName.visited === true &&
+              'Last name is compulsory'
+            }
+            value={state.fields.lastName.value}
+            onChange={(event) => dispatch({ type: 'changeField', fieldName: 'lastName', newValue: event.target.value })}
+          />
+        )}
+        {state.activeStep === 3 && (
+          <NumberInput
+            style={{ width: '300px', paddingTop: '60px' }}
+            id="age"
+            label="Age *"
+            invalid={
+              (!state.fields.age.value || state.fields.age.value.length === 0) && state.fields.age.visited === true
+            }
+            errorText={
+              (!state.fields.age.value || state.fields.age.value.length === 0) &&
+              state.fields.age.visited === true &&
+              'Age is compulsory'
+            }
+            value={state.fields.age.value}
+            onChange={(event) => dispatch({ type: 'changeField', fieldName: 'age', newValue: event.target.value })}
+          />
+        )}
+
+        {state.activeStep === 4 && (
+          <div style={{ marginTop: '20px' }}>
+            <Card border heading="Basic info" headingAriaLevel={3}>
+              <p style={{ margin: 0 }}>First name: {state.fields.firstName.value}</p>
+              <p style={{ margin: 0 }}>Last name: {state.fields.lastName.value}</p>
+              <p style={{ margin: 0 }}>Age: {state.fields.age.value}</p>
+            </Card>
+          </div>
+        )}
+      </div>
+
+      <div
+        style={{
+          height: '100px',
+          display: 'flex',
+          justifyContent: 'flex-start',
+          alignItems: 'flex-start',
+          gap: '24px',
+        }}
+      >
+        <Button
+          disabled={state.activeStep === 1}
+          variant="secondary"
+          onClick={() => dispatch({ type: 'setActive', payload: state.activeStep - 1 })}
+          style={{ height: 'fit-content', width: 'fit-content' }}
+          iconLeft={<IconArrowLeft />}
+        >
+          Previous
+        </Button>
+        <Button
```

Comment:
Just a thought, should the focus move to the erroneous component if validation fails? Now it can be a bit confusing to the user what happened if an error is detected and the next step does not load.

If we follow our own recommendations, we should use an `ErrorSummary` component here to show the error.

Reply:
- [ ] Add ErrorSummary component to show error when trying to move between steps (aka submit a specific step)